homework/index.js

```diff
@@ -30,17 +32,94 @@
+
+  const select = createAndAppend('select', header, { id: 'selector' });
+  const option = x => {
+    for (let i = 0; i < x.length; i++) {
+      createAndAppend('option', select, { text: x[i].name, value: i });
+    }
+  };
+
+  const div = createAndAppend('div', root, { id: 'container' });
+  const leftContainer = createAndAppend('div', div, { class: 'left-block frame' });
+  const table = createAndAppend('table', leftContainer);
+  const listContributor = createAndAppend('tbody', table);
+
+  const rightContainer = createAndAppend('div', div, { class: 'right-block frame' });
+  createAndAppend('p', rightContainer, {
+    text: 'Contributions',
+    class: 'contributor',
+  });
+
+  const list = createAndAppend('ul', rightContainer, { class: 'list1' });
+
+  function renderContributors(selRep) {
+    list.innerHTML = '';
+
+    for (let i = 0; i < selRep.length; i++) {
+      const li = createAndAppend('li', list, { class: 'contributor-item' });
+      const linkFor = createAndAppend('a', li, { target: '_blank', href: selRep[i].html_url });
+      createAndAppend('img', linkFor, { src: selRep[i].avatar_url, class: 'avatar', height: 48 });
+      const divCont = createAndAppend('div', linkFor, { class: 'contributor-data' });
+      createAndAppend('div', divCont, { text: selRep[i].login });
+      createAndAppend('div', divCont, {
+        text: selRep[i].contributions,
+        class: 'badge',
+      });
+    }
+  }
+
+  function renderRepo(listContributors, repo) {
+    const content = [
+      { title: 'Repository', attribute: 'name' },
+      { title: 'Description', attribute: 'description' },
+      { title: 'Forks', attribute: 'forks' },
+      { title: 'Updated', attribute: 'updated_at' },
+    ];
+    listContributors.innerHTML = '';
+
+    for (let i = 0; i < content.length; i++) {
+      const headTitle = createAndAppend('tr', listContributors);
+      createAndAppend('td', headTitle, { text: `${content[i].title} :`, class: 'label' });
+      const cellContent = createAndAppend('td', headTitle);
+      if (content[i].attribute === 'name') {
+        createAndAppend('a', cellContent, {
+          href: repo.html_url,
+          text: repo.name,
+          target: '_blank',
+        });
+      } else {
+        cellContent.textContent = repo[content[i].attribute];
+      }
+    }
+    fetchJSON(repo.contributors_url, (error, contributors) => renderContributors(contributors));
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
+      const root1 = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        createAndAppend('div', root1, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        let optionArr = data;
+        optionArr = optionArr.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
Just FYI - `sort` method here is a mutable method i.e it mutates the array on which it was called. What it means is that if you don't reassign `optionArr` again on line 108, it would still contain the sorted array.

```
const optionArr = data;
optionArr.sort((a,b) => a.name.localeCompare(b.name));

// try console logging optionArr next and verify
console.log(optionArr);
```

Reply:
Ok I delete it ...